contributors/design-proposals/pod-priority-resourcequota.md

```diff
@@ -0,0 +1,79 @@
+  * [Objective](#objective)
+    * [Non-Goals](#non-goals)
+  * [Background](#background)
+  * [Overview](#overview)
+  * [Detailed Design](#detailed-design)
+    * [Expected behavior of ResourceQuota](#resourcequota)
+    * [Expected behavior of ResourceQuota admission controller](#resourcequota-admission-controller)
+
+
+## Objective
+
+This feature is designed to make ResourceQuota become priority aware, 3 sub-tasks are included.
+
+* Add a field to ResourceQuota that specifies priority class names.
+* Incorporate priority in quota checks.
+* ResourceQuota admission controller to check ResourceQuota priority class name and make corresponding decision.
+
+### Non-Goals 
+
+* Add priority in Pod spec (this has been implemented in: #45610)
+
+## Background 
+
+Since we already have [priority field in Pod spec](https://github.com/kubernetes/kubernetes/pull/45610), 
+Pods can now be classified into different priority classes. We would like to be able to create quota for various priority classes in order to manage cluster resources better and limit abuse scenarios. In order to implement this, we need to include priority class name field to ResourceQuota definition.
+
+## Overview 
+
+This design doc introduces a new field in ResourceQuota to specify a group of priority classes for the quota to match with, and explains how quota enforcement logic is changed to apply the quota to pods with the given priority classes.
+
+## Detailed Design 
+
+### Expected behavior of ResourceQuota
+
+The design will be like:
+
+```go
+// ResourceQuotaSpec defines the desired hard limits to enforce for Quota
+type ResourceQuotaSpec struct {
+  // Hard is the set of desired hard limits for each named resource
+  // +optional
+  Hard ResourceList
+  // A collection of filters that must match each object tracked by a quota.
+  // If not specified, the quota matches all objects.
+  // +optional
+  Scopes []ResourceQuotaScope
+  // A list of PriorityClassName with which this quota is expected to match.
+  // If specified, the quota's hard limits are restricted to only account resources oriented around pod which matched with given priority classes.
+  // If not specified, the quota's hard limits are restricted to only account resources oriented around pods without priority class name.
+  // +optional
+  PriorityClassNameList []string
+}
+```
+Any `PriorityClassName` in `PriorityClassNameList` of ResourceQuota will be checked against `PriorityClassName` field in the Pod, in detail:
+
+* If `PriorityClassNameList` is present, for any given `PriorityClassName` in the list:
+    * This quota's hard limits matches to all pods with the matched priority class.
+    * If any `PriorityClassName` is set to *, it means this quota matches all priority classes.
+* If `PriorityClassNameList` is not present, it indicates this `ResourceQuota` matches to pods without priority class name.
+* The absence of `ResourceQuota` for a particular `PriorityClassName` means no quota at that priority.
```

Comment:
absent of a quota for a particular priority class name means the quota is unbounded (unless configured otherwise by operator in admission control configuration)

Reply:
We should definitely make this configurable. I can see that many admins want it to be the other way, i.e., no quota at a priority when there is no explicit ResourceQuota that covers that priority.

Reply:
Right, I amended a new commit to include a configuration example, PTAL